src/main/java/io/druid/data/input/MapBasedRow.java

```diff
@@ -64,26 +62,14 @@ public DateTime getTimestamp()
   @Override
   public List<String> getDimension(String dimension)
   {
-    Object dimValue = event.get(dimension);
+    final Object dimValue = event.get(dimension);
 
     if (dimValue == null) {
-      return Lists.newArrayList();
+      return Collections.emptyList();
     } else if (dimValue instanceof List) {
-      return Lists.transform(
-          (List) dimValue,
-          new Function<Object, String>()
-          {
-            @Override
-            public String apply(Object input)
-            {
-              return String.valueOf(input);
-            }
-          }
-      );
-    } else if (dimValue instanceof Object) {
-      return Arrays.asList(String.valueOf(event.get(dimension)));
+      return Lists.transform((List<Object>) dimValue, TO_STRING_INCLUDING_NULL);
```

Comment:
I left a note in the function implementation, why i did not use Guava's version of it. I can also move the comment up, i dont mind.
And i moved the function definition out, just to have a static instance. Before a new instance is created on each transformation. Does not have to have its own class, though.